internal/provider/project_data_source.go

```diff
@@ -0,0 +1,115 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	unleash "github.com/Unleash/unleash-server-api-go/client"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var (
+	_ datasource.DataSource              = &projectDataSource{}
+	_ datasource.DataSourceWithConfigure = &projectDataSource{}
+)
+
+func NewProjectDataSource() datasource.DataSource {
+	return &projectDataSource{}
+}
+
+type projectDataSource struct {
+	client *unleash.APIClient
+}
+
+type projectDataSourceModel struct {
+	ID          types.String `tfsdk:"id"`
+	Name        types.String `tfsdk:"name"`
+	Description types.String `tfsdk:"description"`
+}
+
+func (d *projectDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*unleash.APIClient)
+	if !ok {
+		tflog.Error(ctx, "Unable to prepare client")
+		return
+	}
+	d.client = client
+
+}
+
+func (d *projectDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_project"
+}
+
+func (d *projectDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Fetch a project definition.",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Description: "The name of this project.",
+				Required:    true,
+			},
+			"id": schema.StringAttribute{
+				Description: "The id of this project.",
+				Required:    true,
```

Comment:
In my datasource I think I used Computed, I'm not sure what's the main difference, but I don't think it makes a huge difference

Reply:
I _think_ required is correct in this place (but what do I know?), because we do actually require the id, it's not an autoincrement id on the unleash side and it's created from a property passed by the create request